### PR TITLE
feat/ CSS code split. Move to vite from rollup.

### PR DESCRIPTION
**Pull request by aluarius**
close: #1545 
close: #1646 
closes https://github.com/epicmaxco/vuestic-ui/issues/1489

## Description
Rollup production build has been fully changed to Vite (`yarn build` to test it). CSS spliting added (for ESM build, including import to separate components).

New average build time is 27 s (vs previous 156 s for my pc).

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

**Comment by aluarius** [comment]
- [ ] delete build:rollup in the end of pr.

File: packages/ui/build/vite/build.mjs
Comment (rustem-nasyrov) [inline packages/ui/build/vite/build.mjs:12]: Use here a single quote.

File: packages/ui/build/vite/build.mjs
Comment (rustem-nasyrov) [inline packages/ui/build/vite/build.mjs:46]:
```suggestion
    const componentName = folderName.split('-').map((el) => el.charAt(0).toUpperCase() + el.slice(1)).join('')
```

File: packages/ui/build/vite/build.mjs
Comment (rustem-nasyrov) [inline packages/ui/build/vite/build.mjs:64]: I think you can check `existsSync(filePath)`, and then inside of the `if` assign new variable and call `renameSync` function.

File: packages/ui/build/vite/build.mjs
Comment (rustem-nasyrov) [inline packages/ui/build/vite/build.mjs:86]: `isEmpty` variable is redundant.

File: packages/ui/build/vite/configs/vite.esm.js
Comment (rustem-nasyrov) [inline packages/ui/build/vite/configs/vite.esm.js:5]: I think here the `format` variable is redundant (in the other configuration files too).

**Review by rustem-nasyrov — CHANGES_REQUESTED** [review]

**Review by RVitaly1978 — APPROVED** [review]

**Comment by m0ksem** [comment]
@aluarius, FYI, please don't merge it ASAP. Let us make some patch releases before this feature will be merged.

I will review it later 👍

**Comment by m0ksem** [comment]
CSS bundles: {
 `vuestic/css` - for cjs, iife - all CSS, for esm - all CSS without components CSS (need to be tested)
      L for esm build components styles must be imported in component
 `vuestic/essential.css` - only essential styles
 `vuestic/components.css` - all component styles without typography, grid, reset etc.
 `vuestic/styles/[typography, grid, reset, etc]` - just css of this pacakges.
}

Faced issue right now. We have component styles only in /css, it makes unable user to use only component styles and essential CSS.

**Review by m0ksem — APPROVED** [review]
I like it! 😊 But let's have it on hold for a while.

**Review by Derranion — APPROVED** [review]

**Comment by aluarius** [comment]
Moved to https://github.com/epicmaxco/vuestic-ui/pull/2081

**Pull request closed without merging**
